### ISSUE TITLE
Poor training performance (3.3% success rate) with pi0 model on locally downloaded official LIBERO dataset

### ISSUE THREAD

**Issue by Alexander-bugwriter** [Physical-Intelligence/openpi#804]
I'm experiencing extremely poor performance when training the pi0 model using PyTorch with a locally downloaded version of the official LIBERO dataset. After training, the model achieves only ~3.3% success rate, which is far below expected performance：

My Personal Training Configuration:
TrainConfig(
    name="pi0_lyh_libero_lora_official_full",
    model=pi0_config.Pi0Config(),
    # Data configuration
    data=LeRobotLiberoDataConfig(
        repo_id="/opt/liblibai-models/user-workspace2/dataset/libero_dataset",
        # local_files_only=True,  # Using local dataset downloaded from physical_intelligence/libero
        base_config=DataConfig(
            prompt_from_task=True,  # Load task descriptions from dataset's task field
        ),
        extra_delta_transform=True,  # Additional transformations required for LIBERO
    ),
    # Load converted PyTorch base model
    weight_loader=weight_loaders.CheckpointWeightLoader(
        "/opt/liblibai-models/user-workspace2/users/lyh/model_checkpoint/pi0/pytorch/pi0_base_lora"
    ),
    # Training hyperparameters
    num_train_steps=50_000,
    batch_size=32,  # Adjusted based on VRAM
    pytorch_training_precision="bfloat16",  # or "float32" if encountering numerical issues
    
    # LoRA specific settings
    # freeze_filter=pi0_config.Pi0Config().get_freeze_filter(),
    # Disable EMA (not needed for LoRA fine-tuning)
    ema_decay=None,
    # Optional: Enable gradient checkpointing to save VRAM
    # gradient_checkpointing=True,
)
Expected Behavior:
The model should achieve success rates comparable to the official pi0 model performance on LIBERO tasks.
Actual Behavior:
Training results in only ~3.3% success rate, which suggests a fundamental issue with the training setup.
Questions/Requests:
Is there an issue with how I'm loading the locally downloaded dataset?
Are there any critical configuration parameters missing or set incorrectly?
Could the LoRA configuration be causing this poor performance?
What is the expected success rate baseline for pi0 on LIBERO, and what training steps/configurations are recommended to achieve it?
Any guidance on debugging this performance issue would be greatly appreciated.

**Comment by yangtiming** [comment]
Same issue, but not 3.3%, pi05_libero pytorch gets much higher result than pi0_libero PyTorch

**Comment by Alexander-bugwriter** [comment]
> Same issue, but not 3.3%, pi05_libero pytorch gets much higher result than pi0_libero PyTorch

Have you tried fine-tuning pi0.5_pytorch on LIBERO tasks? Are you observing that both pi0_pytorch and pi0.5_pytorch models trained with train_pytorch achieve significantly lower performance compared to the numbers reported in the paper and the official JAX checkpoints?
I'm curious whether this is a systemic issue with the entire pi model family's PyTorch pipeline, or if it's specifically a problem with the pi0_pytorch pipeline only.

**Comment by yangtiming** [comment]
My pi0.5_pytorch fine-tuning (batch size 16, 14k iterations) only reaches around 75% on LIBERO tasks.
pi0_pytorch is even worse, only about 50%.

**Comment by BaobaoAndDabao** [comment]
> My pi0.5_pytorch fine-tuning (batch size 16, 14k iterations) only reaches around 75% on LIBERO tasks. pi0_pytorch is even worse, only about 50%.

it is same to me!!

**Comment by BaobaoAndDabao** [comment]
> My pi0.5_pytorch fine-tuning (batch size 16, 14k iterations) only reaches around 75% on LIBERO tasks. pi0_pytorch is even worse, only about 50%.

can you show your train loss?  my train info is grad_norm=0.2106, loss=0.0280, param_norm=1380.0787

**Comment by yangtiming** [comment]
> > My pi0.5_pytorch fine-tuning (batch size 16, 14k iterations) only reaches around 75% on LIBERO tasks. pi0_pytorch is even worse, only about 50%.
> 
> can you show your train loss? my train info is grad_norm=0.2106, loss=0.0280, param_norm=1380.0787

<img width="868" height="321" alt="Image" src="https://github.com/user-attachments/assets/f2015321-3889-4e12-8c23-737d20ae0919" />
<img width="881" height="322" alt="Image" src="https://github.com/user-attachments/assets/7a65b053-522a-44b3-af7c-e0bce16ace27" />

**Comment by xjushengjie** [comment]
Hello, has the problem been resolved? I fine-tuned the original PI0 model with Lora, and the performance is only 0.65. I don't know where the problem lies, maybe because the parameters were not set correctly?
TrainConfig(
        name="pi0_libero_low_mem_finetune",
        model=pi0_config.Pi0Config(paligemma_variant="gemma_2b_lora", action_expert_variant="gemma_300m_lora"),
        data=LeRobotLiberoDataConfig(
            repo_id="pi_csj/libero",
            base_config=DataConfig(prompt_from_task=True),
            extra_delta_transform=True,
        ),
        weight_loader=weight_loaders.CheckpointWeightLoader("gs://openpi-assets/checkpoints/pi0_base/params"),
        num_train_steps=30_000,
        freeze_filter=pi0_config.Pi0Config(
            paligemma_variant="gemma_2b_lora", action_expert_variant="gemma_300m_lora"
        ).get_freeze_filter(),
        ema_decay=None,
    ),

**Comment by HEISENYAN** [comment]
It seems that you are using pi0_torch, in whose conifg, you need to add "pytorch_weight_path" in order to load a pre-trained pi0 model.  "weight_loader" only works for jax version. Otherwise, you are training a pi0 model from 0.

**Comment by hyn1426536274** [comment]
> > > 我的pi0.5_pytorch微调（批次16,14k迭代）在LIBERO任务上只有大约75%。pi0_pytorch更糟，只有大约50%。
> > 
> > 
> > 你能展示你的列车损失吗？我的火车信息是grad_norm=0.2106，损失=0.0280，param_norm=1380.0787
> 
> <img alt="图片" width="868" height="321" src="https://private-user-images.githubusercontent.com/54707723/523408862-f2015321-3889-4e12-8c23-737d20ae0919.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzY0MTgxODQsIm5iZiI6MTc3NjQxNzg4NCwicGF0aCI6Ii81NDcwNzcyMy81MjM0MDg4NjItZjIwMTUzMjEtMzg4OS00ZTEyLThjMjMtNzM3ZDIwYWUwOTE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDE3VDA5MjQ0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlN2ZkOGEyMGZjMDY5MmViYjRiYjdlNTBkZmJjYTc3ZGY0ZWRhOTZlOTA0Mjg5YTIzY2NjNGIwZGIxYmY4YzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.eutSyNtk_Y4049ZSrOgFV0f1-H4-rmjomq8Mm2OcQcs"> <img alt="图片" width="881" height="322" src="https://private-user-images.githubusercontent.com/54707723/523408863-7a65b053-522a-44b3-af7c-e0bce16ace27.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzY0MTgxODQsIm5iZiI6MTc3NjQxNzg4NCwicGF0aCI6Ii81NDcwNzcyMy81MjM0MDg4NjMtN2E2NWIwNTMtNTIyYS00NGIzLWFmN2MtZTBiY2UxNmFjZTI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDE3VDA5MjQ0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkZTU2ODVkOTRjN2FhNjlmNWYzODM1NGFjY2M0NTc4Yjc1MzFhZmZjNjFmNGIxMzAwZmE0MzgzOTczMTJjMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.hoGFEQe-G6y6EFsQYafYl8TFID-bnRXKUKq47ikdjsU">

可以公开你们的train config吗，非常感谢。
